File: src/main/java/Cache.java
import java.time.Duration;
import java.time.LocalDate;
import java.util.BitSet;

public class Cache {
    public LocalDate start, end;
    public BitSet bitSet;
    public Cache(LocalDate start, LocalDate end) {
        this.start = start;
        this.end = end;
        bitSet = new BitSet(offset(start, end));
    }

    public void setDay(LocalDate day) {
        bitSet.set(offset(start, day));
    }

    public void getDay(LocalDate day) {
        bitSet.get(offset(start, day));
    }

    private int offset(LocalDate start, LocalDate end) {
        return (int)Duration.between(start.atStartOfDay(), end.atStartOfDay()).toDays();
    }
}

File: src/main/kotlin/com/example/calendarexperiments/bitset/Cache.kt
package com.example.calendarexperiments.bitset

import java.time.Duration
import java.time.LocalDate
import java.util.*

class Cache(private var start: LocalDate, end: LocalDate) {
    private var bitSet: BitSet = BitSet(offset(start, end))

    fun setDay(day: LocalDate) {
        bitSet.set(offset(start, day))
    }

    fun getDay(day: LocalDate): Boolean {
        return bitSet[offset(start, day)]
    }

    private fun offset(start: LocalDate, end: LocalDate): Int {
        return Duration.between(start.atStartOfDay(), end.atStartOfDay()).toDays().toInt()
    }
}
